### ISSUE TITLE
CPU peaking at 100% and locking the machine

### ISSUE THREAD

**Issue by lucasromanomr** [RocketChat/Rocket.Chat#9636]
### Description:

From time to time the chat rocket is locking a makeup completely, both the node and Mongo causes a machine to have 100% CPU. This is in a small, unused application.

I made the installation use the snap.

### Server Configuration Information:

* Rocket.Chat server version: 0.6.1
* Operating System: Ubunutu 16.04
* Deployment method (snap / docker / tar / etc): snap
* Number of instances running: 1
* DB Replicaset Oplog:
* Version of the node: v8.8.1

### Steps to Reproduce:

1. Start the service
2. uses or does not use
3. expect the problem to happen without doing anything

--- 
*[UPDATE 2]*

In other instance running debian, it also via snap:

<img width="824" alt="captura de tela 2018-02-08 as 12 21 15" src="https://user-images.githubusercontent.com/3859282/35977635-c7a60ea4-0cca-11e8-91ec-3590eab9989f.png">

--- 
*[UPDATE 3]*

A better graphic of the day

<img width="459" alt="captura de tela 2018-02-09 as 12 38 56" src="https://user-images.githubusercontent.com/3859282/36032818-5500424e-0d96-11e8-8f32-bf61d88ca3e9.png">

**Comment by jcatarino** [comment]
I'm having the same problem. I have the same server configuration.

**Comment by geekgonecrazy** [comment]
What kind of user count / activity do you have when this happens?

**Comment by jcatarino** [comment]
I have 50 users, 50% of which using it concurrently and I am using an AWS t2.micro instance.

**Comment by geekgonecrazy** [comment]
My guess is this is mongodb in conjunction with wiredtiger.  On some workloads it seems to hit the CPU very aggressively.  I've opened an issue to track the request to facilitate this change of mongo engine.  #9664 

But unfortunately its not currently possible to change the engine used by mongodb inside of a snap install.

Possible solution would be to throw another core at it.  wiredtiger likes CPU.  

Otherwise... export data and move to a non-snap based install with mongo using mmapv1 instead.

**Comment by lucasromanomr** [comment]
I think I should do a non-snap upgrade so that this is fixed because it's not a Rocket-only server and even though those are not anything at all. The function of the snap is to facilitate and not stop using it to do a manual installation because it is not working.

What I liked the most about him was like this, for using the snap, and having that facility. I hope they solve this problem as soon as possible, we are doing some tests in the company for a possible exchange, but we are running into this processing problem.

**Comment by lucasromanomr** [comment]
Waiting just so we can continue the deployment of the rocket here in the company. So far we will continue with the slack because it is what is working without any problem, and without having to keep on worrying.

**Comment by lucasromanomr** [comment]
I did a clean installation on the same server, a manual installation without using the snap, following steps from the documentation itself. I did not make any configuration any more. The result was that it is now running smoothly, or processing peaks causing the server to crash.

I think it would be interesting to do the check on the installation via snap. For the interesting thing is that people can make an uncomplicated hiring and leave using immediately after.

![5fcc16f892c3881d](https://user-images.githubusercontent.com/3859282/36379459-689110f8-155d-11e8-89cc-99d6e98ad8f0.gif)
